.github/workflows/openvex.yaml

```
@@ -0,0 +1,26 @@
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - 'v*.*.*'
+
+
+jobs:
+  vexctl:
+    runs-on: ubuntu-latest
+    env:
+      TAG: ${{ github.ref_name }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - uses: openvex/generate-vex@31b415924ea0d72ed5f2640f1dee59dea6c2770b
+        name: Run vexctl
+        with:
+          product: pkg:generic/karpenter@${{ env.TAG }}
+          file: /tmp/karpenter.vex.json
+      - name: Upload openvex data
+        env:
+          # Secrets for the workflow can be added at https://github.com/kubernetes-sigs/karpenter/settings/secrets/actions
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh release upload ${{ env.TAG }} /tmp/karpenter.vex.json
```

```suggestion
          gh release upload "$TAG" /tmp/karpenter.vex.json
```

This is safer from a script-injection perspective

---

Should this be part of `release.yaml` directly? I'm realizing now that this is a separate workflow, which is going to have a race with our other release workflow that is triggered on the same event. This should really be a file that we generate after we have already created the release. Or better yet, adding a file to the file block on release creation here: https://github.com/kubernetes-sigs/karpenter/blob/main/.github/workflows/release.yaml#L29

---

Hmm, makes sense. Would moving the `Run vexctl` and `Upload openvex data` steps to after the `Create GitHub release` step in release.yaml be correct?

---

I think if you move the file generation before the actual release creation step, then you can just add the generated file to the `files:` block in the release creation.

---

@jonathan-innis I've made the change. Tried testing the workflow from my repository, but I couldn't get the `Run tejolote` step to work properly. I had to manually create a release for that step to pass, and once the vex json file was also created, the workflow couldn't create a release since I had already created one with the same tag. I believe this wouldn't be an issue when triggering the run from the main repository, although I'd want to run a sanity check for that. The openvex json file is getting generated properly though, as seen in this run [here](https://github.com/sreeram-venkitesh/karpenter/actions/runs/7332501501/job/19966733748). Please let me know what you think!

---

@sreeram-venkitesh Looking at it a little deeper, it looks like the `tejolote` attestation file needs to be generated _after_ the release is created. Testing in my branch, it looks like we need to move this below the release action to get everything to work appropriately.

---

I was able to get this working. There were a few misses. See the changes made here: https://github.com/kubernetes-sigs/karpenter/pull/914